Задание 6. Добавить функциональность регистрации пользователя

Frenchfan opened the pull request:
- убрал большую часть логирования в аспекты
- сделал классы для всех команд фронтэнда с первоначальной реализацией
- отступил от API в части кода 204 при регистрации - возвращаю 201, чтобы показать внутренний id
- не реализованы тестирование, централизованная обработка исключений (сложности с openfeign), но сделаю чуть позже
- уже многовато для одного пулл реквеста - остальное в следующем

Frenchfan committed:
Добавил функциональность регистрации пользователя на уровне фронтэнда - реализация через openfeign (надоел restTemplate)

Frenchfan committed:
Добавил функциональность регистрации пользователя на уровне фронтэнда - реализация через openfeign (надоел restTemplate)

Frenchfan committed:
Добавил функциональность регистрации пользователя на уровне фронтэнда - реализация через openfeign (надоел restTemplate)

Frenchfan committed:
Добавил функциональность регистрации пользователя на уровне фронтэнда - реализация через openfeign (надоел restTemplate)

Frenchfan committed:
Исправил стиль gradle, небольшой рефакторинг обработки комманд

Frenchfan committed:
Исправил стиль gradle, небольшой рефакторинг обработки комманд

Frenchfan committed:
Minor refactoring

Frenchfan committed:
Major project skeleton refacturing, moved MiddleService's address to property, added response values from MiddleService

Frenchfan committed:
исправил ссылку на миддл сервис

Frenchfan committed:
- убрал большую часть логирования в аспекты
- сделал классы для всех команд фронтэнда с первоначальной реализацией
- отступил от API в части кода 204 при регистрации - возвращаю 201, чтобы показать внутренний id
- не реализованы тестирование, централизованная обработка исключений (сложности с openfeign)

Frenchfan committed:
- убрал большую часть логирования в аспекты
- сделал классы для всех команд фронтэнда с первоначальной реализацией
- отступил от API в части кода 204 при регистрации - возвращаю 201, чтобы показать внутренний id
- не реализованы тестирование, централизованная обработка исключений (сложности с openfeign)

Frenchfan committed:
- убрал большую часть логирования в аспекты
- сделал классы для всех команд фронтэнда с первоначальной реализацией
- отступил от API в части кода 204 при регистрации - возвращаю 201, чтобы показать внутренний id
- не реализованы тестирование, централизованная обработка исключений (сложности с openfeign) - уже многовато

Frenchfan committed:
Merge branch 'trunk' into feature/6

vaganov-vadim commented on .github/workflows/gradle.yml:
```diff
@@ -32,12 +32,14 @@ jobs:
     - name: Make gradlew executable
       run: chmod +x ./gradlew
 
+#проверить, можно ли не использовать build
```
Давай попробуем clean добавить? Такое ощущение, что у тебя какой-то build cache застрял там.

Frenchfan replied:
не помогло

Frenchfan committed:
- убрал большую часть логирования в аспекты
- сделал классы для всех команд фронтэнда с первоначальной реализацией
- отступил от API в части кода 204 при регистрации - возвращаю 201, чтобы показать внутренний id
- не реализованы тестирование, централизованная обработка исключений (сложности с openfeign) - уже многовато

Frenchfan committed:
Merge remote-tracking branch 'origin/feature/6' into feature/6

# Conflicts:
#	src/main/java/com/gpb/sumkintelegrambot/web/MiddleServiceClient.java

vaganov-vadim commented on src/main/java/com/gpb/sumkintelegrambot/configuration/Command.java:
```diff
@@ -4,7 +4,9 @@ public enum Command {
     START("/start", "стартовое сообщение"),
```
Не уверен в необходимости enum, мы это всё можем хранить в самих командах. А так получается лишняя зависимость между классами.

Frenchfan replied:
Это скорее для того, чтобы точно не забыть обработчик команды сделать. Я бы все-таки оставил

vaganov-vadim replied:
А что значит не забыть?) Команда = её обработчик

vaganov-vadim commented on src/main/java/com/gpb/sumkintelegrambot/web/MiddleServiceClient.java:
```diff
@@ -1,13 +1,38 @@
+    ResponseEntity<RegistrationDto> registerUser(@RequestBody UserDto userDto);
+
+    @GetMapping("/v2/users/{id}")
+    UUID getUserById(@PathVariable Long id);
```
На будущее - есть смысл для всех возвращаемых значений на стороне миддла предусматривать какую-то DTO'шку, даже если там одно значение. Сегодня только UUID, завтра может добавиться что-то ещё. Добавить необязательный параметр в DTO - неломающее изменение, заменить строчку на объект - ломающее.

Frenchfan replied:
Учел, поменял feign. Кроме того, добавил контейнеризацию (Dockerfile + docker-compose), обработку исключений feign для регистрации, провел рефакторинг

vaganov-vadim reviewed (changes requested):
Привет! В целом всё круто, разве что добавил пару комментов. По CI - надо разобраться, что не так, можно просто попробовать на других ветках поиграть. Если там воспроизводится, то проблема в самом воркфлоу.

Frenchfan committed:
- убрал большую часть логирования в аспекты
- сделал классы для всех команд фронтэнда с первоначальной реализацией
- отступил от API в части кода 204 при регистрации - возвращаю 201, чтобы показать внутренний id
- не реализованы тестирование, централизованная обработка исключений (сложности с openfeign) - уже многовато

Frenchfan committed:
- убрал большую часть логирования в аспекты
- сделал классы для всех команд фронтэнда с первоначальной реализацией
- отступил от API в части кода 204 при регистрации - возвращаю 201, чтобы показать внутренний id
- не реализованы тестирование, централизованная обработка исключений (сложности с openfeign) - уже многовато

Frenchfan committed:
- добавил валидацию
- добавил dto для всех ответов от миддла

Frenchfan committed:
- добавил валидацию
- добавил dto для всех ответов от миддла

Frenchfan committed:
-добавил контейнеризацию

Frenchfan committed:
-добавил контейнеризацию

Frenchfan committed:
-добавил контейнеризацию

Frenchfan committed:
-добавил контейнеризацию

Frenchfan committed:
-добавил обработку feignException для связи с backend
- небольшой рефакторинг

Frenchfan committed:
-добавил обработку feignException для связи с backend
- небольшой рефакторинг

Frenchfan committed:
- добавлена реализация Get для User и соответствующие Dto

Frenchfan committed:
- небольшой рефакторинг

vaganov-vadim reviewed (approved):
Я всё ещё не согласен с Enum, но в целом аппрув :)
